app/src/main/java/com/arctouch/codechallenge/api/TmdbApi.kt

```diff
@@ -1,40 +1,73 @@
 package com.arctouch.codechallenge.api
 
-import com.arctouch.codechallenge.model.GenreResponse
+import com.arctouch.codechallenge.BuildConfig
+import com.arctouch.codechallenge.api.endpoints.Tmdb
+import com.arctouch.codechallenge.model.Genre
 import com.arctouch.codechallenge.model.Movie
-import com.arctouch.codechallenge.model.UpcomingMoviesResponse
 import io.reactivex.Observable
-import retrofit2.http.GET
-import retrofit2.http.Path
-import retrofit2.http.Query
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
 
-interface TmdbApi {
+object TmdbApi {
+    private val api: Tmdb by lazy {
+        val logging = HttpLoggingInterceptor()
+        logging.level = HttpLoggingInterceptor.Level.BODY
 
-    companion object {
-        const val URL = "https://api.themoviedb.org/3/"
-        const val API_KEY = "1f54bd990f1cdfb230adb312546d765d"
-        const val DEFAULT_LANGUAGE = "pt-BR"
-        const val DEFAULT_REGION = "BR"
+        val httpClient = OkHttpClient.Builder()
+        httpClient.addInterceptor(logging)
+        httpClient.addInterceptor{chain ->
+            val origin = chain.request()
+            val originalHttpUrl = origin.url()
+
+            val url = originalHttpUrl.newBuilder()
+                    // if the api were sent at post, it could be inserted
+                    // as an interceptor at this point
+                    .build()
+
+            val requestBuilder = origin.newBuilder().url(url)
+            val request = requestBuilder.build()
+            chain.proceed(request)
+        }
+
+        val retrofit = Retrofit.Builder()
+                .baseUrl(BuildConfig.URL)
+                .client(OkHttpClient.Builder().build())
```

Comment:
The OkHttp? Because he build the requisition, and and inserts other parameters the request, such as timeout, for example. Also treats the HTTP response for kotlin in this case.